File: frontend/src/app/common/model/currency-type.ts
export enum CurrencyType {
  USD = 'USD',
  PLN = 'PLN',
  EUR = 'EUR',
  GBP = 'GBP',
  CNY = 'CNY',
  MYR = 'MYR',
  CAD = 'CAD',
  AUD = 'AUD',
  INR = 'INR',
  BRL = 'BRL',
  NOK = 'NOK',
}

export namespace CurrencyType {
  export const Symbols: Record<CurrencyType, string> = {
    [CurrencyType.USD]: '$',
    [CurrencyType.PLN]: 'zł',
    [CurrencyType.EUR]: '€',
    [CurrencyType.GBP]: '£',
    [CurrencyType.CNY]: '¥',
    [CurrencyType.MYR]: 'RM',
    [CurrencyType.CAD]: 'C$',
    [CurrencyType.AUD]: 'A$',
    [CurrencyType.INR]: '₹',
    [CurrencyType.BRL]: 'R$',
    [CurrencyType.NOK]: 'kr',
  };
}

File: backend/common/src/main/java/pl/muybien/enums/CurrencyType.java
package pl.muybien.enums;

public enum CurrencyType {
    USD("$"),
    PLN("zł"),
    GBP("£"),
    EUR("€"),
    CNY("¥"),
    MYR("RM"),
    CAD("C$"),
    AUD("A$"),
    INR("₹"),
    BRL("R$"),
    NOK("kr");

    private final String symbol;

    CurrencyType(String symbol) {
        this.symbol = symbol;
    }

    public String getSymbol() {
        return this.symbol;
    }
}
